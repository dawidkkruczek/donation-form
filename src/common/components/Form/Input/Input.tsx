import React, { ComponentProps, ReactElement } from 'react';
import classNames from 'classnames';
import styles from './Input.module.scss';

export interface Props extends ComponentProps<'input'> {}

function Input({
  className,
  type = 'text',
  ...inputProps
}: Props): ReactElement {
  return (
    <input
      className={classNames(styles.root, className)}
      type={type}
      {...inputProps}
    />
  );
}

export default Input;
