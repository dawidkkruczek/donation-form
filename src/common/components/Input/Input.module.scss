.root {
  background: var(--color-white);
  border-radius: 4px;
  border: 1px solid var(--color-stroke);
  color: var(--color-purple-gray);
  font-family: 'Rubik';
  font-size: 24px;
  font-weight: 500;
  line-height: 1.2;
  padding: 16px;
  transition: all 0.5s ease;
  width: 100%;

  &:focus,
  &:focus-visible {
    border-color: var(--color-midnight-purple);
    outline: none;
  }

  &::placeholder {
    color: var(--color-placeholder);
  }

  &:-webkit-autofill {
    -webkit-background-clip: text;
  }

  &::-webkit-datetime-edit-day-field:focus,
  &::-webkit-datetime-edit-month-field:focus,
  &::-webkit-datetime-edit-year-field:focus {
    background: var(--color-midnight-purple);
    color: var(--color-white);
  }
}
