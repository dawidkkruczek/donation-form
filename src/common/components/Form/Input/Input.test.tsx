import React from 'react';
import { render } from '@testing-library/react';
import Input from './Input';

describe('common/components/Input', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Input />);
    expect(asFragment()).toMatchSnapshot();
  });
});
