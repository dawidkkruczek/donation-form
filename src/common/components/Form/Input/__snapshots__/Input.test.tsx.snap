// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/components/Input should render correctly 1`] = `
<DocumentFragment>
  <input
    class="root"
    type="text"
  />
</DocumentFragment>
`;
